### DISCUSSION TITLE
don't generate pseudo-class in tailwind? i get warnings when using with princexml

### DISCUSSION THREAD

**Discussion by deadcoder0904** [tailwindlabs/tailwindcss#9549] [Help]
i am using tailwind with princexml.

since it's a paged media, some pseudo-classes are not supported. i get warnings like:

```bash
prince: output/styles/tailwind.css:80: warning: unknown pseudo-class ':where()'
prince: output/styles/tailwind.css:358: warning: unknown pseudo-element '::placeholder'
prince: output/styles/tailwind.css:507: warning: unknown pseudo-element '::backdrop'
prince: output/styles/tailwind.css:80: warning: unknown pseudo-class ':where()'
prince: output/styles/tailwind.css:358: warning: unknown pseudo-element '::placeholder'
prince: output/styles/tailwind.css:507: warning: unknown pseudo-element '::backdrop'
prince: output/styles/tailwind.css: warning: unsupported properties: resize
```

how do i disable those warnings? is it possible?

**Comment by wongjn** [comment]
According to the [Prince XML command line documentation](https://www.princexml.com/doc/command-line/#logging-options), you should be able to disable these warnings with the `--no-warn-css` command line flag.

**Reply by deadcoder0904** [reply]
what about not generating them altogether?

because those warnings are necessary to know the bug.

recently, i found `@page:first` isn't working thanks to those warning. i changed it to `@page :first` to make it work.

i wouldn't have figured it out without the warning.

**Reply by wongjn** [reply]
Tailwind core preflight styles use `:where()`, `::placeholder` and `resize` so this core plugin would have to be disabled to remove the warnings pertaining to the pseudo class, the pseudo element and the property:
```js
module.exports = {
  corePlugins: {
    preflight: false,
  },
};
```
However, this is Tailwind's styling reset so you would most likely have to modify your project's styling to accommodate this plugin being disabled. Also, you may need to check any other Tailwind plugins/custom code that could be using  `:where()`, `::placeholder` and `resize`.

For `::backdrop`, this is most likely coming from the rules that are used to set CSS variables defaults for some utilities. You could try setting the `optimizeUniversalDefaults` experimental flag to remove it:
```js
module.exports = {
  experimental: {
    optimizeUniversalDefaults: true,
  },
};
```
But be warned – it is experimental!

**Reply by deadcoder0904** [reply]
alas, i thought there was a way to disable it somehow with a flag. i did google & found nothing so thought i'd ask. i'll just keep it as it is then. thank you :)
